docs/overview.md

```diff
@@ -1,26 +1,116 @@
+
+For example, consider a Markdown link like `[some text](#a-label)`. In MyST Markdown, this defines a **cross-reference** to `#a-label`, but it uses Markdown link syntax. We use a MyST Transformer to convert that Markdown to a cross-reference AST node like so:
+
+- First parse the Markdown `[some text](#a-label)`.
+- The result is a MyST AST node for a **Markdown link**.
+- Next, search the document AST for any Markdown link nodes with a target that starts with `#`. Check if there's [a target](./cross-references.md) for the link. If so, it is meant to be a cross reference.
+- If so, run a **Transformer** that converts the Markdown Link node into a Cross Reference node.
+
+Some other uses for Transformers include:
+
+- Lifting metadata from `code-cells` to their parent structures
+- Check that figures have alt-texts
+- Convert non-standard AST nodes (e.g., ones generated from a custom user directive) into ones that MyST knows how to render[^ex-transform-node].
+
+[^ex-transform-node]: This is a pattern used in e.g. https://github.com/projectpythia-mystmd/cookbook-gallery/blob/main/pythia-gallery.py where an `executable transform` (non-JS transform that communicates over `STDIO` with `JSON`) takes custom `pythia-cookbooks` nodes and converts them (via some HTTP fetches) to a grid of cards by outputting the relevant grid and card AST nodes.
+
+See [](#develop:transforms) for more details about how transforms operate on an AST and what intermediate steps look like.
+
+(overview-themes)=
+## How do MyST themes and templates work?
+
+MyST {term}`themes` and MyST {term}`templates` are both related to _exporting_ a resolved MyST AST into one or more final output formats. They differ in the amount of _customizability_ that a user has in altering their behavior, but the end-result is the same (generating an export artifact).
+
+```{glossary}
+Templates
+: Templates are usually used for [static exports](./documents-exports.md) (like LaTeX or Typst). The are structured like a final document (e.g., a LaTeX template for a specific journal), but with fields that MyST uses to insert document metadata at export time (e.g., a `doc.author` field if you want to insert the author name). See [the `myst-templates` GitHub repository](https://github.com/myst-templates) for a list of templates that you can use and contribute to.
+
+Themes
+: Themes are a special-case of templates because they expose more customizability than a typical template does. They are usually meant for _websites_, because there is a lot more control a website exposes compared to a static document like LaTeX. Themes have more complex logic, and thus their source-code is often in a dedicated repository. For examples, see the [build outputs of the `book` and `article` web themes are `myst-templates`](https://github.com/myst-templates#templates), but their [source code is in `myst-theme`](https://github.com/jupyter-book/myst-theme/tree/main/themes).
```

Comment:
Are themes special cases of templates, or are they just templates? And if they're just templates, why do we have a special name for them?

Sounds like "theme == web template".